src/pages/AnimeOnsen/main.ts

```diff
@@ -2,70 +2,111 @@ import { pageInterface } from '../pageInterface';
 
 export const AnimeOnsen: pageInterface = {
   name: 'AnimeOnsen',
-  domain: ['https://animeonsen.xyz', 'https://www.animeonsen.xyz'],
+  domain: 'https://animeonsen.xyz',
   database: 'AnimeOnsen',
   languages: ['English', 'Japanese'],
   type: 'anime',
   isOverviewPage(url) {
-    // added from: https://github.com/MALSync/MALSync/pull/984#discussion_r770401087
+    // check if current page is details/overview page
+    const [, page] = new URL(url).pathname.split('/');
+    if (/^details$/i.test(page)) return true;
     return false;
   },
   isSyncPage(url) {
-    // check if current page is /watch page
-    const { pathname, searchParams } = new URL(url);
-    if (pathname.startsWith('/watch') && searchParams.has('v')) return true;
+    // check if current page is watch/sync page
+    const [, page] = new URL(url).pathname.split('/');
+    if (/^watch$/i.test(page)) return true;
     return false;
   },
+  overview: {
+    getTitle(_) {
+      // get anime name
+      return j.$('div.metadata-container div.title span[lang="en"]').text() || '';
+    },
+    getIdentifier(url) {
+      // get animeonsen content id for database
+      const [, , contentId] = new URL(url).pathname.split('/');
+      return contentId || '';
+    },
+    uiSelector(selector) {
+      const wrapper = document.createElement('div');
+      wrapper.classList.add('malp-wrapper');
+      wrapper.innerHTML = j.html(selector);
+      j.$('div.content-details').after(j.html(wrapper.outerHTML));
+    },
+    getMalUrl(provider) {
+      // get myanimelist anime url
+      return new Promise(resolve => {
+        if (provider === 'MAL') resolve(j.$('meta[name="ao-content-mal-url"]').attr('content') || false);
+        else resolve(false);
+      });
+    },
+    list: {
+      offsetHandler: false,
+      elementsSelector() {
+        return j.$('div.episode-list > a');
+      },
+      elementUrl(selector) {
+        return utils.absoluteLink(selector.attr('href'), AnimeOnsen.domain);
+      },
+      elementEp(selector) {
+        return Number($(selector).find('div.episode').data('episode'));
+      },
+    },
+  },
   sync: {
-    getTitle(url) {
+    getTitle(_) {
       // get anime name
-      return j.$('meta[name="ao-api-malsync-title"]').attr('value') || '';
+      return j.$('span.ao-player-metadata-title').text();
     },
     getIdentifier(url) {
-      // get ao.id identifier for database
-      const urlParams = new URL(url).searchParams;
-      const identifier = urlParams.get('v');
-      return identifier || '';
+      // get animeonsen content id for database
+      const [, , contentId] = new URL(url).pathname.split('/');
+      return contentId || '';
     },
     getOverviewUrl(url) {
       // generate ao.details url
-      const urlParams = new URL(url).searchParams;
+      const [, , contentId] = new URL(url).pathname.split('/');
```

Comment:
The contentId is retrived 3 times exactly the same way. I would recommend to just call the overview.getIdentifier directly.
`AnimeOnsen.overview!.getIdentifier(url);`